src/aiai_eval/named_entity_recognition.py

```diff
@@ -251,35 +365,67 @@ def _prepare_predictions_and_labels(
                 predictions and labels will be used for all the metrics.
         """
 
-        if id2label is not None:
-            # Remove ignored index (special tokens)
-            predictions = [
-                [
-                    id2label[pred_id]
-                    for pred_id, lbl_id in zip(pred, label)
-                    if lbl_id != -100
+        # In case we have received predictions and labels which are already converted
+        # from ids to label tags, we do not need to convert them again.
+        # This is the case when the model is a spacy model.
+        if any(
+            predictions_np.dtype == dtype
+            for dtype in {
+                np.float16,
+                np.float32,
+                np.float64,
+                np.int16,
+                np.int32,
+                np.int64,
+            }
+        ) and any(
+            labels_np.dtype == dtype
+            for dtype in {
+                np.float16,
+                np.float32,
+                np.float64,
+                np.int16,
+                np.int32,
+                np.int64,
+            }
+        ):
```

Comment:
Maybe make two functions in `utils.py` which checks whether a NumPy array contains integers and floats, respectively? Since we use this in the `task` module as well.